waku/waku_rln_relay/rln_relay.nim

```diff
@@ -211,25 +213,30 @@ proc validateMessage*(
   # track message count for metrics
   waku_rln_messages_total.inc()
 
-  # checks if the `msg`'s epoch is far from the current epoch
-  # it corresponds to the validation of rln external nullifier
-  # get current rln epoch
-  let epoch: Epoch = rlnPeer.getCurrentEpoch()
+  # checks if the message's timestamp is within acceptable range
+  let currentTime = Timestamp(getTime().toUnixFloat())
 
-  let
-    msgEpoch = proof.epoch
-    # calculate the gaps
-    gap = absDiff(epoch, msgEpoch)
-
-  trace "epoch info", currentEpoch = fromEpoch(epoch), msgEpoch = fromEpoch(msgEpoch)
-
-  # validate the epoch
-  if gap > rlnPeer.rlnMaxEpochGap:
-    # message's epoch is too old or too ahead
-    # accept messages whose epoch is within +-MaxEpochGap from the current epoch
-    warn "invalid message: epoch gap exceeds a threshold",
-      gap = gap, payloadLen = msg.payload.len, msgEpoch = fromEpoch(proof.epoch)
-    waku_rln_invalid_messages_total.inc(labelValues = ["invalid_epoch"])
+  var timeDiff = uint64(currentTime - msg.timestamp)
+
+  debug "time info",
+    currentTime = currentTime, messageTime = msg.timestamp, msgHash = msg.hash
+
+  if msg.timestamp > currentTime:
```

Comment:
The only validity requirement is that the `msg.timestamp` should not differ from ours by more than the max allowed gap. This can be both into the future or into the past, to cover normal clock jitter.